server/utils/csv/csvProcess.js

```diff
@@ -0,0 +1,61 @@
+const mongoose = require("mongoose");
+const multer = require('multer');
+const csv = require('csv-parser');
+const fs = require('fs');
+
+const { validateProspectInput } = require("../../validation/prospect");
+const Prospect = require("../../models/Prospect");
+
+async function processCsvData(data) {
+    let skippedCount = 0;
+    let successCount = 0;
+    let uploadArray = [];
+
+    let emailsFromCsv = data
+          .filter(data => validateProspectInput(data).isValid)
+          .map(data => data.email);
+    emailsFromCsv = new Set(emailsFromCsv);
+
+    let dbEmails = await Prospect.find({}, {email:1, _id:0});
```

Comment:
Looks great -  this API call should be only prospects owned by this user, and you should only get emails that are from the CSV to make it more efficient - check out in query (https://docs.mongodb.com/manual/reference/operator/query/in/)